### DISCUSSION TITLE
Loss function scaling

### DISCUSSION THREAD

**Discussion by chao1224** [mir-group/nequip#259] [Q&A] [answered]
Hi there,
Thank you for providing this nice code base.
I have a detailed question on the latest config when trying to reproduce the MD17 results.

According to #131 , the latest normalization for v0.5 is:
```
per_species_rescale_scales_trainable: false
per_species_rescale_shifts_trainable: false
per_species_rescale_shifts: dataset_per_atom_total_energy_mean
per_species_rescale_scales: dataset_forces_rms
```

This is saying that we will do scaling and shifting for each atom-level energy prediction without learnable paraemeters. The corresponding eq10 will be udpated to $\hat E = \sum_i \sigma \hat E_i + \lambda$ .

Then we suppose that $E, F_i$ are the ground-truth total energy and atom-level force. (1) Without global shift, the loss is $\mathcal{L}(\sigma E, \hat E) + 1000 \cdot \mathcal{L}(\sigma F_i, \nabla \hat E_i)$,  is this correct? (2) Or if the scaling is not needed for the ground-truth values, i.e., $\mathcal{L}(E, \hat E) + 1000 \cdot \mathcal{L}(F_i, \nabla \hat E_i)$?

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @chao1224:

There are basically two kinds of units in `nequip`:

  1. "Normalized" units, where any global rescaling of the outputs (`RescaleOutputs` / `RescaleEnergyEtc`) has not yet been applied
  2. "Real" units, which the dataset was provided in

Per-species / per-atom scaling/shifting is applied before any global scaling/shifting.  The loss function is computed in normalized units using ground-truth targets that are un-scaling into normalized units.
